src/Mvc/test/WebSites/SimpleWebSiteWithWebApplicationBuilder/Program.cs

```diff
@@ -49,3 +49,10 @@ public class MyController : ControllerBase
     [HttpGet("/greet")]
     public string Greet() => $"Hello human";
 }
+
+namespace SimpleWebSiteWithWebApplicationBuilder
+{
+    public partial class Program
```

Comment:
Isn't this effectively FakeStartup? Perhaps leaving this as-is is better since there is a pending ask from the compiler team to be able to access the top level type by name.

@halter73 thoughts?

Reply:
It is in a way, but this is one of the two patterns I've seen suggested to be used for WAF. The other one is to use `InternalsVisibleTo`.

I figured this made the tests use that same patterns users are being steered towards.

I agree that having the compiler let you do this through a setting rather than through code would be nicer though.